File: modules/expo-logger/ios/ExpoLoggerCore.m
#import "ExpoLoggerCore.h"
#import <os/log.h>

@interface ExpoLoggerCore()
@property (nonatomic, assign) int logCount;
@property (nonatomic, strong) os_log_t logger;
@end

@implementation ExpoLoggerCore

+ (instancetype)shared {
    static ExpoLoggerCore *sharedInstance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[self alloc] init];
    });
    return sharedInstance;
}

- (instancetype)init {
    if (self = [super init]) {
        _logCount = 0;
        _logger = os_log_create("com.newarchspike", "ExpoLogger");
    }
    return self;
}

// Manual reset for development/testing
- (void)resetCount {
    _logCount = 0;
}

// ========================================
// BUSINESS LOGIC
// ========================================

- (void)logInfo:(NSString *)message {
    os_log_with_type(self.logger, OS_LOG_TYPE_INFO, "%{public}@", message);
    self.logCount++;
}

- (void)logWarning:(NSString *)message {
    os_log_with_type(self.logger, OS_LOG_TYPE_DEFAULT, "%{public}@", message);
    self.logCount++;
}

- (void)logError:(NSString *)message {
    os_log_with_type(self.logger, OS_LOG_TYPE_ERROR, "%{public}@", message);
    self.logCount++;
}

// ========================================
// STATE MANAGEMENT
// ========================================

- (int)getLogCount {
    return self.logCount;
}

@end

File: modules/expo-logger/ios/ExpoLoggerCore.swift
import Foundation
import os.log

/**
 * ExpoLoggerCore - Pure native business logic (NO React Native dependencies)
 * 
 * This class owns the logger state and contains all business logic.
 * It can be called from:
 * - ExpoLoggerModule (Expo Module for JavaScript)
 * - ModuleInterop (for other native modules like cg-webview)
 * - Unit tests (no RN runtime needed)
 */
public class ExpoLoggerCore {
    public static let shared = ExpoLoggerCore()
    
    private var logCount: Int = 0
    private let logger: os.Logger
    
    private init() {
        self.logger = os.Logger(subsystem: "com.newarchspike", category: "ExpoLogger")
    }
    
    // Manual reset for development/testing
    public func resetCount() {
        logCount = 0
    }
    
    // ========================================
    // BUSINESS LOGIC
    // ========================================
    
    public func logInfo(_ message: String) {
        logger.info("\(message, privacy: .public)")
        logCount += 1
    }
    
    public func logWarning(_ message: String) {
        logger.warning("\(message, privacy: .public)")
        logCount += 1
    }
    
    public func logError(_ message: String) {
        logger.error("\(message, privacy: .public)")
        logCount += 1
    }
    
    // ========================================
    // STATE MANAGEMENT
    // ========================================
    
    public func getLogCount() -> Int {
        return logCount
    }
}

